src/Symfony/Component/Translation/LocaleSwitcher.php

```diff
@@ -11,23 +11,31 @@
 
 namespace Symfony\Component\Translation;
 
+use Symfony\Component\Routing\RequestContext;
 use Symfony\Contracts\Translation\LocaleAwareInterface;
 
 /**
  * @author Kevin Bond <kevinbond@gmail.com>
  */
-final class LocaleSwitcher implements LocaleAwareInterface
+class LocaleSwitcher implements LocaleAwareInterface
```

Comment:
Because the class is registered as an autowiring alias, it means we're telling ppl they can type-hint for it.
Making the class `final` breaks any SOLID principles, so I removed the keyword.

Reply:
Imho, we should register `LocaleAwareInterface` as an autowiring alias, and not advocate typehinting implementations.

Reply:
That works for get/setLocale, but runWithLocale isn't in any interfaces, and I'm not sure we want to add one for it.
I'm fine type-hinting with the implem personnaly, but only if it's open for inheritance of course.